tachyon/zk/plonk/examples/fibonacci/fibonacci1_circuit.h

```diff
@@ -0,0 +1,172 @@
+#ifndef TACHYON_ZK_PLONK_EXAMPLES_FIBONACCI_FIBONACCI1_CIRCUIT_H_
+#define TACHYON_ZK_PLONK_EXAMPLES_FIBONACCI_FIBONACCI1_CIRCUIT_H_
+
+#include <array>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "tachyon/zk/plonk/constraint_system/circuit.h"
+
+namespace tachyon::zk::plonk {
+
+template <typename F>
+struct AssignedRow {
+  AssignedCell<F> a;
+  AssignedCell<F> b;
+  AssignedCell<F> c;
+
+  AssignedRow() = default;
+  AssignedRow(AssignedCell<F>&& a, AssignedCell<F>&& b, AssignedCell<F>&& c)
+      : a(std::move(a)), b(std::move(b)), c(std::move(c)) {}
+};
+
+template <typename F>
+struct Fibonacci1Config {
+  using Field = F;
+
+  Fibonacci1Config(std::array<AdviceColumnKey, 3> advice, Selector selector,
+                   const InstanceColumnKey& instance)
+      : advice(std::move(advice)), selector(selector), instance(instance) {}
+
+  Fibonacci1Config Clone() const {
+    return Fibonacci1Config(advice, selector, instance);
```

Comment:
It's optional, but it looks more simple.

```suggestion
    return {advice, selector, instance};
```